benchmarks/advection_stabilization/advection_stabilization.cc

```diff
@@ -0,0 +1,170 @@
+        const double f_r = A*r + B/r;
+        const double g_r = A*r/2 + B*std::log(r)/r + C/r;
+        const double v_r = g_r*k*sin(k*theta);
+        const double v_theta = f_r*cos(k*theta);
+        const double v_x = cos(theta)*v_r - sin(theta)*v_theta;
+        const double v_y = sin(theta)*v_r + cos(theta)*v_theta;
+        return Point<2> (v_x,v_y);
+      }
+
+      double
+      Annulus_pressure (const Point<2> &pos,
+                        const double k)
+      {
+        const double x = pos[0];
+        const double y = pos[1];
+        const double r=std::sqrt(x*x+y*y);
+        const double theta = std::atan2(y,x);
+        const double f_r = 2*r + B/r;
+        const double g_r = A*r/2 + B*std::log(r)/r + C/r;
+        const double h_r=(2*g_r-f_r)/r;
+        return k*h_r*sin(k*theta)+rho_0*gravity*(outer_radius-r);
+      }
+
+      template<int dim>
+      double
+      Annulus_normal_traction (const Point<dim> &pos,
+                               const double k)
+      {
+        Assert (dim == 2, ExcNotImplemented());
+        const double x = pos[0];
+        const double y = pos[1];
+        const double r=std::sqrt(x*x+y*y);
+        const double theta = std::atan2(y,x);
+        const double f_r = 2*r + B/r;
+        const double g_r = A*r/2 + B*std::log(r)/r + C/r;
+        return k * 3.*f_r/r * std::sin(k*theta) - rho_0 * g_r * (outer_radius - r);
+      }
+
+    }
+  }
+}
+
+
+namespace aspect
+{
+  namespace PrescribedStokesSolution
+  {
+    using namespace dealii;
+
+    /**
+     * A class that implements a circular, divergence-free flow field
+     * around the origin of the coordinate system.
+     *
+     * @ingroup PrescribedStokesSolution
+     */
+    template <int dim>
+    class AdvectionBenchmark : public Interface<dim>, public SimulatorAccess<dim>
+    {
+      public:
+
+        virtual
+        void stokes_solution (const Point<dim> &p, Vector<double> &value) const;
+    };
+  }
+}
+
+namespace aspect
+{
+  namespace PrescribedStokesSolution
+  {
+    template <int dim>
+    void AdvectionBenchmark<dim>::stokes_solution (const Point<dim> &, Vector<double> &) const
+    {
+      return;
+    }
+
+
+    template <>
+    void AdvectionBenchmark<2>::stokes_solution (const Point<2> &p, Vector<double> &value) const
+    {
+      Tensor<1,2> velocity = AnnulusBenchmark::AnalyticSolutions::Annulus_velocity (p, 4);
+      value(0) = velocity[0];
+      value(1) = velocity[1];
+
+      double pressure = AnnulusBenchmark::AnalyticSolutions::Annulus_pressure (p, 4);
+      value(2) = pressure;       // pressure
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace AnnulusBenchmark
+  {}
+  namespace PrescribedStokesSolution
+  {
+    ASPECT_REGISTER_PRESCRIBED_STOKES_SOLUTION(AdvectionBenchmark,
+                                               "advection benchmark",
+                                               "This value describes a vector field that rotates "
```

Comment:
Modify this explanation to state that this benchmark prescribed the Stokes solution of the annulus benchmark.